woocommerce/woocommerce
# [Experimental] Move ProductType store to shared folder

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR moves the `ProductType` [store](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/store) folder to the [shared](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce-blocks/assets/js/shared) directory.

The `ProductType` store is currently used by the `Add to Cart with Options` block. However, we now need to use this store in other places, such as the `Stock Indicator` block.

Closes #53851.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install and activate the `WooCommerce Beta Tester` plugin
2. Go to `WooCommerce Admin Test Helper` -> `Features` and enable `add-to-cart-with-options-stepper-layout` and `blockified-add-to-cart`.
3. Smoke test the `Add to Cart with Options (Experimental)` block.
4. Create a new product.
5. Go to Pages > Add New Page.
6. Add a `Single product` block and select the product you just created.
7. Remove the current _Add to Cart with Options_ block.
8. Everything should work well
9. Now go to the Sincle Product template and add an `Add to Cart with Options (Experimental)` block.
10. Verify that after adding the block the `Type switcher` selector is now visible in the Inspector > Template.
11. Confirm the selected product type is consistent.

https://github.com/user-attachments/assets/0c56abdb-fc03-416c-b893-b4cca89049ca



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
